PictureApp/Controller/PicturesViewController.swift

```diff
@@ -0,0 +1,306 @@
+//
+//  ViewController.swift
+//  PictureApp
+//
+//  Created by Irina Gorbachenko on 26.11.2021.
+//
+
+import UIKit
+import NetworkLayer
+
+class PicturesViewController: UIViewController, OptionsViewControllerDelegate,FilterViewControllerDelegate {
+    
+    var totalHits = 0
+    var currentPage = 1
+    var selectedCategoryForSort = [String]()
+    var orderForCategories : Order = .popular
+    var collectionViewFlowLayout : UICollectionViewFlowLayout!
+    private let collectionView = UICollectionView (frame: .zero, collectionViewLayout: UICollectionViewFlowLayout())
+    
+    var hits = [Post](){
+        didSet{
+            DispatchQueue.main.async{
+                self.collectionView.reloadData()
+            }
+        }
+    }
+    
+    
+    // MARK:Delegate for categories
+    func filterViewController(_ controller: FilterViewController, didSelectCategory selectedCategory: [String]) {
+         currentPage = 1
+         hits.removeAll()
+         fetchHits(order: orderForCategories, selectedCategory: selectedCategory,currentPage: currentPage)
+          DispatchQueue.main.async{
+            self.collectionView.reloadData()
+        }
+        navigationController?.popViewController(animated: true)
+        selectedCategoryForSort = selectedCategory
+    }
+    
+    
+    //MARK:Delegate for order
+    func optionsViewController(_ controller: OptionsViewController, didSelectOrderStrategy order: Order) {
+        currentPage = 1
```

Comment:
These lines used it optionsViewController method and in filterViewController method. Suggest to extract them to some method and use it instead of these lines.
```
currentPage = 1
hits.removeAll()
```

Reply:
Fixed